lib/active_merchant/billing/gateways/d_local.rb

```diff
@@ -155,11 +159,22 @@ def parse_house_number(address)
 
       def add_card(post, card, action, options = {})
         post[:card] = {}
+        if card.is_a?(NetworkTokenizationCreditCard)
+          post[:card][:network_token] = card.number
+          post[:card][:cryptogram] = card.payment_cryptogram
+          post[:card][:eci] = card.eci
+          # used case of Network Token: 'CARD_ON_FILE', 'SUBSCRIPTION', 'UNSCHEDULED_CARD_ON_FILE'
+          post[:card][:stored_credential_type] = options[:stored_credential_type] if options[:stored_credential_type]
```

Comment:
I think we will want to utilize the existing stored credential hash structure, and map those values to what is expected here.

Reply:
Yes, you are right, I will use the hash structure.